src/alchemlyb/preprocessing/subsampling.py

```diff
@@ -34,63 +87,198 @@ def slicing(df, lower=None, upper=None, step=None, force=False):
+
+    def random_selection(name, group):
+        tried = set()
+        while True:
+            group_c, selection = _how_random(name, group, tried=tried)
+            try:
+                indices, calculated = subsample(group_c, group)
+            except:
+                tried.add(selection)
+            else:
+                break
+
+        return indices, calculated
+
+    for name, group in data.groupby(level=index_names):
+
+        if not force and _check_multiple_times(group):
+            raise KeyError("Duplicate time values found; decorrelation "
+                           "is only meaningful for a single, contiguous, "
+                           "and sorted timeseries")
+
+        if column is not None:
+            if isinstance(column, pd.Series):
+                group_c = column.groupby(level=index_names).get_group(name)
+                indices, calc = subsample(group_c, group)
+            elif column in group.columns:
+                group_c = group[column]
+                indices, calc = subsample(group_c, group)
+        elif how is not None:
+            if (how == 'right') or (how == 'left'):
+                try:
+                    group_c = _how_lr(name, group, how)
+                except KeyError:
+                    indices, calc = random_selection(name, group)
+                else:
+                    indices, calc = subsample(group_c, group)
+            elif how == 'random':
+                indices, calc = random_selection(name, group)
+            elif how == 'sum':
+                group_c = _how_sum(name, group)
+                indices, calc = subsample(group_c, group)
+            else:
+                raise ValueError("`how` cannot be '{}';"
+                " see docstring for available options".format(how))
+
+        group_s = slicing(group, lower=lower, upper=upper, step=step)
+        resdata.append(group_s.iloc[indices])
+
+        if return_calculated:
+            calculated[name] = calc
+    
+    if return_calculated:
+        return pd.concat(resdata), pd.DataFrame(calculated)
+    else:
+        return pd.concat(resdata)
+
+
+def statistical_inefficiency(data, column=None, how=None, lower=None,
+        upper=None, step=None, conservative=True, return_calculated=False,
+        force=False, random_state=None):
+    """Subsample an alchemlyb DataFrame based on the calculated statistical
+    inefficiency of one of its columns.
+
+    Calculation of statistical inefficiency and subsequent subsampling will be
+    performed separately on groups of rows corresponding to each set of lambda
+    values present in the DataFrame's index. Each group will be sorted on the
+    outermost (time) index prior to any calculation.
+
+    The `how` parameter determines the observable used for calculating the
+    correlations within each group of samples. The options are as follows:
+
+        'right'
+            The recommended setting for 'u_nk' datasets; the column immediately
```

Comment:
Who's doing the recommending and on what grounds?

If my student used alchemlyb and said "I used the recommended settings" then I will ask "recommendation based on what?" and then the student won't know what to say.

I'd like a more scholarly approach to recommendations. Or don't recommend anything.